test/message/_message.cxx

```diff
@@ -1,223 +1,232 @@
+auto RunPushPullWithAlignment(string const& transport, string const& _address) -> void
+{
+    ProgOptions config;
+    config.SetProperty<string>("session", tools::Uuid());
+    auto factory(TransportFactory::CreateTransportFactory(transport, tools::Uuid(), &config));
 
-    FairMQChannel push{"Push", "push", factory};
+    Channel push{"Push", "push", factory};
+    Channel pull{"Pull", "pull", factory};
+    auto const address(tools::ToString(_address, "_", transport));
     push.Bind(address);
-
-    FairMQChannel pull{"Pull", "pull", factory};
     pull.Connect(address);
 
-    size_t alignment = 64;
-
-    FairMQMessagePtr outMsg1(push.NewMessage(100, fair::mq::Alignment{alignment}));
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(outMsg1->GetData()) % alignment, 0);
-    ASSERT_EQ(push.Send(outMsg1), 100);
-
-    FairMQMessagePtr inMsg1(pull.NewMessage(fair::mq::Alignment{alignment}));
-    ASSERT_EQ(pull.Receive(inMsg1), 100);
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(inMsg1->GetData()) % alignment, 0);
-
-    FairMQMessagePtr outMsg2(push.NewMessage(32, fair::mq::Alignment{alignment}));
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(outMsg2->GetData()) % alignment, 0);
-    ASSERT_EQ(push.Send(outMsg2), 32);
-
-    FairMQMessagePtr inMsg2(pull.NewMessage(fair::mq::Alignment{alignment}));
-    ASSERT_EQ(pull.Receive(inMsg2), 32);
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(inMsg2->GetData()) % alignment, 0);
+    {
+        Alignment const align{64};
+        for (size_t const size : {100, 32}) {
+            auto outMsg(push.NewMessage(size, align));
+            auto inMsg(pull.NewMessage(align));
+
+            ASSERT_TRUE(CheckMsgAlignment(*outMsg, align));
+            ASSERT_EQ(push.Send(outMsg), size);
+            ASSERT_EQ(pull.Receive(inMsg), size);
+            ASSERT_TRUE(CheckMsgAlignment(*inMsg, align));
+        }
+    }
 
-    FairMQMessagePtr outMsg3(push.NewMessage(100, fair::mq::Alignment{0}));
-    ASSERT_EQ(push.Send(outMsg3), 100);
+    {
+        Alignment const align{0};
+        size_t const size{100};
+        auto outMsg(push.NewMessage(size, align));
+        auto inMsg(pull.NewMessage(align));
 
-    FairMQMessagePtr inMsg3(pull.NewMessage(fair::mq::Alignment{0}));
-    ASSERT_EQ(pull.Receive(inMsg3), 100);
+        ASSERT_EQ(push.Send(outMsg), size);
+        ASSERT_EQ(pull.Receive(inMsg), size);
+    }
 
-    FairMQMessagePtr msg1(push.NewMessage(25));
-    msg1->Rebuild(50, fair::mq::Alignment{alignment});
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(msg1->GetData()) % alignment, 0);
+    for (auto const align : {Alignment{64}, Alignment{32}}) {
+        size_t const size25{25};
+        size_t const size50{50};
 
-    size_t alignment2 = 32;
-    FairMQMessagePtr msg2(push.NewMessage(25, fair::mq::Alignment{alignment}));
-    msg2->Rebuild(50, fair::mq::Alignment{alignment2});
-    ASSERT_EQ(reinterpret_cast<uintptr_t>(msg2->GetData()) % alignment2, 0);
+        auto msg(push.NewMessage(size25));
```

Comment:
Actually this is wrong. Before the test did (1) Rebuild from default alignment to specific alignment, (2) Rebuild from specific alignment to a different specific alignment.

Reply:
I'll fix it in my own branch, as I am extending some of the tests here.

Reply:
Thx